### PR TITLE
adds fixtures by Id endpoint, model controller. Tested successfully

### PR DESCRIPTION
**Pull request by ASP73**
adds fixtures by Id endpoint, model controller. Tested successfully

### CODE REVIEW

File: back-end/models/fixtures_byId.model.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+JOIN
+    teams t1 ON f.team1_id = t1.team_id
+JOIN
+    teams t2 ON f.team2_id = t2.team_id
+JOIN
+    venues v ON f.match_venue = v.venue_id WHERE fixture_id = $1;`,
+      [fixture_id]
+    )
+    .then(({ rows }) => {
+      console.log(rows.length);
```
Comment (miors) [inline back-end/models/fixtures_byId.model.js]: console.log needs to be removed

File: back-end/controllers/fixtures_byId.controller.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const { fetchFixturesById } = require("../models/fixtures_byId.model.js");
+
+const getFixturesById = (req, res, next) => {
+  const { fixture_id } = req.params;
+  fetchFixturesById(fixture_id)
+    .then((fixture) => res.status(200).send({ fixture }))
+    .catch((err) => {
```
Comment (miors) [inline back-end/controllers/fixtures_byId.controller.js]: We capture the error in app.js, so you only need .catch(next) instead

File: back-end/controllers/fixtures_byId.controller.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+      if (err.status && err.message) {
```
Comment (miors) [inline back-end/controllers/fixtures_byId.controller.js]: ditto the above as we capture the error in app.js

File: back-end/__tests__/app.test.js [hunk 4]
```diff
@@ -240,3 +240,47 @@ describe("/api/teams", () => {
...
   });
 });
+
+describe("/api/fixtures/:fixture_id", () => {
+  test("GET 200: Responds with endpoint json data", () => {
+    return request(app)
+      .get("/api/fixtures/3")
+      .expect(200)
+      .then(({ body }) => {
+        const fixture = body.fixture;
```
Comment (miors) [inline back-end/__tests__/app.test.js]: for clarity` fixture` needs to be `fixtures`

**Review by miors — CHANGES_REQUESTED** [review]
Tests are working. Only a few requests for change particularly the errors are supposed to be passed to api.js from controller and a few minor issues

File: back-end/controllers/fixtures_byId.controller.js (hunk 2 above)
Comment (ASP73) [reply back-end/controllers/fixtures_byId.controller.js]: amended

File: back-end/controllers/fixtures_byId.controller.js (hunk 3 above)
Comment (ASP73) [reply back-end/controllers/fixtures_byId.controller.js]: amended

File: back-end/models/fixtures_byId.model.js (hunk 1 above)
Comment (ASP73) [reply back-end/models/fixtures_byId.model.js]: removed

**Review by miors — APPROVED** [review]
LGTM. Great work @ASP73 👍

**Pull request merged**